### ISSUE TITLE
Regarding the length and width of rectangular training yolov5, specify them separately in dlstreamer

### ISSUE THREAD

**Issue by jayer95** [dlstreamer/dlstreamer#336]
The training data of my project is all 4K (3840x2160, landscape), so I use rectangular training during training, as shown below:

![train_batch0](https://user-images.githubusercontent.com/35683311/204428754-8a42c7bd-60a6-4985-8780-a51281c881dc.jpg)

I refer to the previous Issue to complete the conversion of the yolov5 model,
https://github.com/dlstreamer/dlstreamer/issues/322

After the training is complete, I use the following command on [yolov5](https://github.com/ultralytics/yolov5/blob/master/export.py) to convert the model to ONNX, as follows:

```
python export.py --weights yolov5n-license_plate_best.pt --imgsz 384 640 --simplify --include onnx
```

Next, I installed OpenVINO 2022.2 python package, and converted to IR model using the following command:

```
mo  --input_model yolov5n-license_plate_best.onnx --model_name yolov5n-license_plate_best --scale 255 --reverse_input_channels --output onnx::Reshape_329,onnx::Reshape_367,onnx::Reshape_405 --data_type FP16 --output_dir yolov5/FP16
```

I got 3 files of IR model:
- yolov5n-license_plate_best.bin
- yolov5n-license_plate_best.mapping
- yolov5n-license_plate_best.xml

The following is the dlstreamer json I wrote:

```
{
    "json_schema_version": "2.2.0",
    "input_preproc": [
        {
            "params": {
                "resize": "aspect-ratio",
                "color_space": "BGR"
            }
        }
    ],
    "output_postproc": [
        {
            "converter": "yolo_v5",
            "output_sigmoid_activation": true,
            "do_cls_softmax": true,
            "iou_threshold": 0.5,
            "bbox_number_on_cell": 3,
            "cells_number_x": 12,
            "cells_number_y": 20,
            "classes": 1,
            "anchors": [
                9.0,
                5.0,
                10.0,
                6.0,
                12.0,
                7.0,
                14.0,
                8.0,
                17.0,
                9.0,
                20.0,
                11.0,
                23.0,
                13.0,
                27.0,
                15.0,
                33.0,
                18.0
            ],
            "masks": [
                6,
                7,
                8,
                3,
                4,
                5,
                0,
                1,
                2
            ],
	    "labels": ["license plate"]
        }
    ]
}
```

When I do yolov5 inference in dlstreamer, I need to keep it consistent with the "letterbox" algorithm used during training, so I specify in json: **"resize": "aspect-ratio"** to maintain the original aspect ratio of the license plate.

And separate the length and width to specify the cell number as:
**"cells_number_x": 12**,
**"cells_number_y": 20**,

**"cells_number_x": 12** is the height, which is 2160 and resized into a 640x640 square letterbox with a size of 360, but 360 cannot be divisible by 32, so add the gray edge becomes 384,

384/32=12

**"cells_number_x": 20** is the height, which is 3840 and resized into a 640x640 square letterbox with a size of 640,

640/32=20

Finally, I run the following commands under the installed dlstreamer (Ubuntu 20.04):
The resolution of 4K_demo.mp4 is 3840x2160,

```
gst-launch-1.0 filesrc location="4K_demo.mp4" ! decodebin ! gvadetect model="/home/yolov5/yolov5n-license_plate_best_ir/FP16/yolov5n-license_plate_best.xml" model_proc="yolov5n-license_plate_best.json" device=CPU ! queue ! gvawatermark ! gvafpscounter ! videoconvert ! avenc_mpeg4 ! mp4mux ! filesink location="4K_demo_output.mp4"
```
    
4K_demo_output.mp4 has incorrect results, the bounding-boxes appears but the position is incorrect, as shown below:

![gnome-shell-screenshot-BAVHW1](https://user-images.githubusercontent.com/35683311/204433506-a04de8d0-e743-4cfc-9416-65b6be9a776b.png)

May I ask which step I made a mistake, or has dlstreamer not yet optimized this part?

**Comment by vidyasiv** [comment]
Hello @jayer95 ,
Will it be possible for you to share the input media and the yolov5n-license_plate_best.pt file?

**Comment by jayer95** [comment]
@vidyasiv
Sure, please refer to the link below:
https://drive.google.com/file/d/1aJ7gZHPiGZd7DQZb1_I_yN8XQsAMS-ug/view?usp=sharing

I will provide a test model .zip, the dendrogram is as follows:

```unzip dlstreamer_yolov5_rect_debug.zip
```
> dlstreamer_yolov5_rect_debug
> ├── 4K_demo.mp4
> ├── 4K_demo_pytorch_yolov5_result_log.txt
> ├── 4K_demo_pytorch_yolov5_result.mp4
> ├── yolov5n-license_plate_best
> │   └── FP16
> │       ├── yolov5n-license_plate_best.bin
> │       ├── yolov5n-license_plate_best.mapping
> │       └── yolov5n-license_plate_best.xml
> ├── yolov5n-license_plate_best.json
> ├── yolov5n-license_plate_best.onnx
> └── yolov5n-license_plate_best.pt

yolov5n-license_plate_best.pt is tested on yolov5 v6.2 and above, and the result is as expected. Please refer to "4K_demo_pytorch_yolov5_result_log.txt" and "4K_demo_pytorch_yolov5_result.mp4":

```
python detect.py \
    --source 4K_demo.mp4 \
    --weights yolov5n-license_plate_best.pt \
    --imgsz 640 \
    --conf-thres 0.5 \
    --iou-thres 0.1
```

Thank you Intel for your help.

**Comment by jayer95** [comment]
@vidyasiv please let me know if there is any new progress, thanks.

**Comment by kkoryun** [comment]
Hi @jayer95. Is it possible to change/transpose/swapaxes the outputs shapes of the model from 1x18x12x20 to 1x18x20x12 (...40x24 and ...80x48 for corresponding layers)? Where number of cells on X (20,40,80) is at position 2, number of cells on Y in at position 3. Due to not quite correct output processing, the postprocessing handle only the dims described above.

**Comment by jayer95** [comment]
@kkoryun 
So the bugs is that the dimension order of the model output is incorrect? Does it need to do model export reshape?

**Comment by akwrobel** [comment]
Hi @jayer95 This is a bug, and will be resolved in our 2023.0 release (forthcomming)

**Comment by jayer95** [comment]
@akwrobel 

Ok, thanks.
